pkg/packet/bgp/prefix_sid.go

```diff
@@ -540,3 +543,135 @@ func (s *SRv6SIDStructureSubSubTLV) String() string {
 		s.TranspositionOffset,
 	)
 }
+
+func NewPathAttributePrefixSID(psid *api.PrefixSID) (*PathAttributePrefixSID, error) {
+	t := BGP_ATTR_TYPE_PREFIX_SID
+	s := &PathAttributePrefixSID{
+		PathAttribute: PathAttribute{
+			Flags: PathAttrFlags[t],
+			Type:  t,
+		},
+		TLVs: make([]PrefixSIDTLVInterface, 0),
+	}
+	for _, raw := range psid.Tlvs {
+		var tlv ptypes.DynamicAny
+		if err := ptypes.UnmarshalAny(raw, &tlv); err != nil {
+			return nil, err
+		}
+		switch v := tlv.Message.(type) {
+		case *api.SRv6L3ServiceTLV:
+			tlvLength, tlvs, err := UnmarshalSubTLVs(v.SubTlvs)
+			if err != nil {
+				return nil, err
+			}
+			o := &SRv6L3ServiceAttribute{
+				TLV: TLV{
+					Type:   TLVType(5),
+					Length: tlvLength,
+				},
+			}
+			s.PathAttribute.Length += tlvLength
+			// Storing Sub TLVs in a Service TLV
+			o.SubTLVs = append(o.SubTLVs, tlvs...)
+			// Adding Service TLV to Path Attribute TLV slice.
+			s.TLVs = append(s.TLVs, o)
+		default:
+			return nil, fmt.Errorf("unknown or not implemented Prefix SID type: %+v", v)
+		}
+	}
+	// Final Path Attribute Length is 3 bytes of the header and 1 byte Reserved1
+	s.PathAttribute.Length += (3 + 1)
+	return s, nil
+}
+
+func UnmarshalSubTLVs(stlvs map[uint32]*api.SRv6TLV) (uint16, []PrefixSIDTLVInterface, error) {
+	p := make([]PrefixSIDTLVInterface, 0)
```

Comment:
it is possible to give here a pre-allocation hint, like:
```
p := make([]PrefixSIDTLVInterface, 0, len(stlvs))
```

Reply:
make sense, done..